semantic/inference_test.go

```diff
@@ -1988,6 +2002,13 @@ r.a < r.b`,
 				},
 			},
 		},
+		{
+			name: "access nonexistant object",
+			script: `r = {a: 1}
+r.b.c + r.a
+`,
+			wantErr: errors.New("missing object properties (b)"),
```

Comment:
We discussed in slack but it would be nice to have a similar test case for properties that resolve to regexp or function types. We might have to skip them for right now and create a specific issue to unskip them later.